src/ServiceControl.Monitoring.AcceptanceTests/Tests/When_querying_queue_length_data.cs

```diff
@@ -16,37 +17,83 @@ class When_querying_queue_length_data : AcceptanceTest
+                         ec.EnableMetrics()
+                             .SendMetricDataToServiceControl(global::ServiceControl.Monitoring.Settings.DEFAULT_ENDPOINT_NAME, TimeSpan.FromSeconds(1));
+                     });
+                     c.DoNotFailOnErrorMessages();
+                     c.When(async s =>
+                     {
+                         for (var i = 0; i < 10; i++)
+                         {
+                             await s.SendLocal(new SampleMessage());
+                         }
+                     });
+                 })
+                 .WithEndpoint<SendingEndpoint>(c =>
+                 {
+                     c.CustomConfig(ec =>
+                     {
+                         ec.MakeInstanceUniquelyAddressable("2");
+                         ec.EnableMetrics()
+                             .SendMetricDataToServiceControl(global::ServiceControl.Monitoring.Settings.DEFAULT_ENDPOINT_NAME,
+                             TimeSpan.FromSeconds(1),
+                             metricsInstanceId.ToString("N"));
+                     });
+                     c.DoNotFailOnErrorMessages();
+                     c.When(async s =>
+                     {
+                         for (var i = 0; i < 10; i++)
+                         {
+                             await s.SendLocal(new SampleMessage());
+                         }
+                     });
+                 })
+                 .Done(async c =>
+                 {
+                     var result = await this.TryGet<MonitoredEndpointDetails>($"/monitored-endpoints/{endpointName}");
+
+                     if (!result.HasResult)
+                     {
+                         return false;
+                     }
+
+                     monitoredEndpointDetails = result.Item;
 
-                    metricReported = result.HasResult && result.Items[0].Metrics["queueLength"].Average > 0;
+                     instance1 = monitoredEndpointDetails.Instances.SingleOrDefault(instance => instance.Id == instanceId.ToString("N"));
+                     instance2 = monitoredEndpointDetails.Instances.SingleOrDefault(instance => instance.Id == metricsInstanceId.ToString("N"));
 
-                    if (metricReported)
-                    {
-                        c.TestEnded.SetResult(true);
-                    }
+                     if (instance1 == null || instance2 == null)
+                     {
+                         return false;
+                     }
 
-                    return metricReported;
-                })
-                .Run();
+                     if (monitoredEndpointDetails.Digest.Metrics["queueLength"].Average == 0.0)
+                     {
+                         return false;
+                     }
 
+                     c.TestEnded.SetResult(true);
 
-            Assert.IsTrue(metricReported);
+                     return true;
+                 })
+                 .Run();
```

Comment:
Don't we need `Assert.Pass()` here?

Reply:
Test blows up and fails if there is a timeout so it will only be green if average > 0.0 ?